src/checkers/inference/InferrableAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,33 @@
+package checkers.inference;
+
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+
+/**
+ * Interface for indicating that the current the annotatedTypeFactory wish to be
+ * used by checker-framework-inference. This interface allows a
+ * annotatedTypeFactory to configure its own inference behavior.
+ */
+
+public interface InferrableAnnotatedTypeFactory {
+
+    /**
+     * This method could be called if the TreeAnnotator defined in current type
```

Comment:
I'm not sure if the TreeAnnotator returned by this method is allowed to also add \@VarAnnot.